owlsim-services/src/main/java/org/monarchinitiative/owlsim/services/resources/OntologyMatchResource.java

```diff
@@ -61,6 +52,28 @@
         return matches;
     }
 
+    
     // TODO - API for comparing two entities
 
+    @GET
+    @Path("/{entity}/{ontology}")
+    @Timed
+    @CacheControl(maxAge = 2, maxAgeUnit = TimeUnit.HOURS)
+    @ApiOperation(value = "Match", response = MatchSet.class,
+    notes = "Additional notes on the match resource.")
+    public List<SimpleClassMatch> getEntityMatches(
+            @ApiParam(value = "entity, e.g. MP:0001951",
+            required = true) @PathParam("entity") String entity,
+            @ApiParam(value = "ontology to be matched, e.g. HP",
+            required = true) @PathParam("ontology") String ontology)
+                    throws UnknownFilterException, IncoherentStateException {
+    	
+    	// TODO it's weird to resolve the curie here
```

Comment:
Ahh, no sorry - I'm mistaken. The ```curieUtil``` is used internally within the ```BMKnowledgebase```, but is not accessible from it. Providing the ClassMatcher with the ```curieUtil``` and ```bmKnowledgeBase``` in the constructor would solve this. Then you can move

```java
String resolveEntity = curieUtil.getIri(entity).orElse(entity); 
```
into the ```ClassMatcher```.

I just wonder to what extent we need to hide the ```CurieUtil``` inside the ```BMKnowledgeBase``` - presumably we're always going to use the OWLAPI and the ```CurieUtil``` in the ```BMKnowledgeBase```? If that's the case and these are going to be required externally to query the ```BMKnowledgeBase```, then it should expose them so they can be used, or provide methods like:
'''
public String resolveIri(String entity)
    return curieUtil.getIri(entity).orElse(entity); 
```